gocommon/SystemConfig.go

```diff
@@ -240,7 +240,7 @@ func (sys *SystemConfig) GetURL(name ServiceName) (string, error) {
 
 	url := addr + EndpointPrefixes[name]
 	if !(strings.HasPrefix(url, "http://") || strings.HasPrefix(url, "https://")) {
-		url = DefaultOutboundProtocol + url
+		url = DefaultOutboundProtocol + "://" + url
 	}
```

Comment:
Why not use `net/url`?

    u, _ := url.Parse(addr + EndpointPrefixes[name])
    if u.Scheme == "" {
      u.Scheme = DefaultOutboundProtocol
    }

    // actual URL string can be accessed using u.String()

It's more idiomatic and needs less prefix stuff and string concatenation.

Reply:
I like the idea but this happens a couple of different places and opens up the scope of this fix.  I would say if we wanted to make a change like this, a separate ticket would need to be opened to change all the places we do this string concatenation. (two places in this file and one similar spot in nt.go where we also check for the prefix)

Reply:
Plus, I'd rather make as few changes as possible if we're going to be pushing this fix to UC-Stable